nix/ext/pg_cron.nix

```diff
@@ -1,31 +1,147 @@
+      sed -e "/^default_version =/d" \
+          -e "s|^module_pathname = .*|module_pathname = '\$libdir/pg_cron'|" \
+          pg_cron.control > pg_cron--${pgCronVersion}.control
+    '';
+
+    installPhase = ''
+      mkdir -p $out/{lib,share/postgresql/extension,bin}
+      
+      # Install versioned library
+      install -Dm755 pg_cron${postgresql.dlSuffix} $out/lib/pg_cron-${pgCronVersion}${postgresql.dlSuffix}
+      
+      # Install version-specific files
+      install -Dm644 pg_cron--${pgCronVersion}.sql $out/share/postgresql/extension/
+      install -Dm644 pg_cron--${pgCronVersion}.control $out/share/postgresql/extension/
+      
+      # Install upgrade scripts
+      find . -name 'pg_cron--*--*.sql' -exec install -Dm644 {} $out/share/postgresql/extension/ \;
+    '';
   };
 
+  getVersions = pg:
+    if lib.versionAtLeast pg.version "17"
+    then { "1.6.4" = allVersions."1.6.4"; }
+    else allVersions;
+
+  allVersionsForPg = lib.mapAttrs mkPgCron (getVersions postgresql);
+
+in
+stdenv.mkDerivation {
+  pname = "pg_cron-all";
+  version = versionString;
+
+  buildInputs = lib.attrValues allVersionsForPg;
+
+  dontUnpack = true;
+  dontConfigure = true;
+  dontBuild = true;
+
   installPhase = ''
-    mkdir -p $out/{lib,share/postgresql/extension}
+    mkdir -p $out/{lib,share/postgresql/extension,bin}
+    
+    # Install all versions
+    for drv in ${lib.concatStringsSep " " (lib.attrValues allVersionsForPg)}; do
+      ln -sv $drv/lib/* $out/lib/
+      cp -v --no-clobber $drv/share/postgresql/extension/* $out/share/postgresql/extension/ || true
+    done
+    
+    # Create default symlinks
+    latest_control=$(ls -v $out/share/postgresql/extension/pg_cron--*.control | tail -n1)
+    latest_version=$(basename "$latest_control" | sed -E 's/pg_cron--([0-9.]+).control/\1/')
+    
+    # Create main control file with default_version
+    echo "default_version = '$latest_version'" > $out/share/postgresql/extension/pg_cron.control
+    cat "$latest_control" >> $out/share/postgresql/extension/pg_cron.control
+    
+    # Library symlink
+    ln -sfnv pg_cron-$latest_version${postgresql.dlSuffix} $out/lib/pg_cron${postgresql.dlSuffix}
+
+    # Create version switcher script
+    cat > $out/bin/switch_pg_cron_version <<'EOF'
+    #!/bin/sh
+    set -e
+
+    if [ $# -ne 1 ]; then
+      echo "Usage: $0 <version>"
+      echo "Example: $0 1.4.2"
+      exit 1
+    fi
+
+    VERSION=$1
+    LIB_DIR=$(dirname "$0")/../lib
+
+    # Use platform-specific extension
+    if [ "$(uname)" = "Darwin" ]; then
+      EXT=".dylib"
+    else
+      EXT=".so"
+    fi
+
+    # Check if version exists
+    if [ ! -f "$LIB_DIR/pg_cron-$VERSION$EXT" ]; then
+      echo "Error: Version $VERSION not found"
+      exit 1
+    fi
+
+    # Update library symlink
+    ln -sfnv "pg_cron-$VERSION$EXT" "$LIB_DIR/pg_cron$EXT"
+
+    echo "Successfully switched pg_cron to version $VERSION"
+    EOF
```

Comment:
Are other extensions going to do the same thing? If so, could there be a "generic" wrapper that could have `substituteAll` be called on?

Reply:
There may be one other extension that does something similar but I am not sure yet if it will be generic enough to re-use